--- r6p2/src/devicedrv/mali/linux/mali_memory.c	2019-03-19 13:23:39.740056345 +0100
+++ r6p2/src/devicedrv/mali/linux/mali_memory.c	2019-03-19 13:25:17.284057010 +0100
@@ -57,10 +57,14 @@
 	vma->vm_private_data = NULL;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
-static int mali_mem_vma_fault(struct vm_fault *vmf)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 17, 0)
+    static vm_fault_t mali_mem_vma_fault(struct vm_fault *vmf)
 #else
-static int mali_mem_vma_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
+    #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+    static int mali_mem_vma_fault(struct vm_fault *vmf)
+    #else
+    static int mali_mem_vma_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
+    #endif
 #endif
 {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
